### DISCUSSION TITLE
How to modify the output name?

### DISCUSSION THREAD

**Discussion by hanyuu11** [pybind/pybind11#4540] [Q&A] [answered]
I have a module named rl_cpp, and this is the cmakelist:
```
pybind11_add_module(rl_cpp SHARED pybind.cc)
set_target_properties(rl_cpp PROPERTIES OUTPUT_NAME "rl_cpp")
```
The output file's name is still rl_cpp.cp38-win_amd64.pyd, is the part ".cp38-win_amd64" unchangeable? I want the file name to be "rl_cpp.pyd". Are there any methods?

**Comment by sebe324** [comment, marked as answer]
Hello!
I know I'm responding to this quite late, but I've recently had the same problem, but after some digging I got this to work.
What worked for me was using the Python_add_library function istead of pybind11_add_module.
[Link](https://pybind11.readthedocs.io/en/stable/cmake/index.html#new-findpython-mode)

You can try this:

```
find_package(Python COMPONENTS Interpreter Development)

Python_add_library(rl_cpp pybind.cc)
target_link_libraries(rl_cpp PRIVATE pybind11::headers)
set_target_properties(rl_cpp PROPERTIES
                                INTERPROCEDURAL_OPTIMIZATION ON
                                CXX_VISIBILITY_PRESET ON
                                VISIBILITY_INLINES_HIDDEN ON
                                OUTPUT_NAME rl_cpp)
```

Here is the result on my pc:

![image](https://github.com/pybind/pybind11/assets/58781463/e5544b17-8f5f-449c-8c48-b6e4d6f9fd4f)

**Reply by hanyuu11** [reply]
It works on my PC. Thanks for your answer!

**Reply by sebe324** [reply]
no problem :)
